flatten/flatten_1.rb

```diff
@@ -0,0 +1,21 @@
+#flatten.rb
+
+def flatten(array)
+  if array.empty?
+    []
+  else
+    out=[]
+    array.each do |value|
```

Comment:
each is functional http://www.tutorialspoint.com/ruby/ruby_iterators.htm

This is the way you should iterate through your arrays in ruby